functions/src/index.ts

```diff
@@ -232,6 +232,123 @@ function cleanAverages(averages: SensorReading[]): CleanedReadings {
   return new CleanedReadings(latitude, longitude, cleanedAverages);
 }
 
+/**
+ * Computes the AQI for PM 2.5 given the appropriate AQI breakpoints. in most use cases, this function will be called from a function which knows those breakpoints.
+ *
+ * Adapted from EPA function available at https://www.airnow.gov/sites/default/files/custom-js/conc-aqi.js
+ *
+ * @param pm25Concentration - the PM 2.5 concentration to calculate the AQI for
+ * @param lowConcentrationBreakpoint - the low breakpoint for PM 2.5 that the concentration falls within
+ * @param highConcentrationBreakpoint - the high breakpoint for PM 2.5 that the concentration falls within
+ * @param lowIndexBreakpoint - the low breakpoint for AQI that the AQI will fall between
+ * @param highIndexBreakpoint - the high breakpoint for AQI that the AQI will fall between
+ */
+function _indexCalculation(
+  highIndexBreakpoint: number,
+  lowIndexBreakpoint: number,
+  highConcentrationBreakpoint: number,
+  lowConcentrationBreakpoint: number,
+  pm25Concentration: number
+): number {
+  const indexRange = highIndexBreakpoint - lowIndexBreakpoint;
+  const concentrationRange =
+    highConcentrationBreakpoint - lowConcentrationBreakpoint;
+  const rangeRelativeConcentration =
+    pm25Concentration - lowConcentrationBreakpoint;
+
+  return (
+    (indexRange / concentrationRange) * rangeRelativeConcentration +
+    lowIndexBreakpoint
+  );
+}
+
+/**
+ * Computes the AQI for a given PM 2.5 concentration in micrograms per cubic meter.
+ *
+ * Adapted from EPA functions available at https://www.airnow.gov/sites/default/files/custom-js/conc-aqi.js
+ * @param pm25Conc - the PM 2.5 concentration to calculate the AQI for
```

Comment:
You use pm25Concentration as a variable name above and I think it's fine that way, so nit: rename to pm25Concentration or pm25 (I think it communicates the same thing as long as you add the units to the doc comment)
Nit: add the units of concentration to the param too